test/apps/iodemo/io_demo.cc

```diff
@@ -1678,7 +1679,7 @@ class DemoClient : public P2pDemoCommon {
         while (((_num_sent - _num_completed) > max_outstanding) &&
                (_status == OK)) {
             if ((count++ < BUSY_PROGRESS_COUNT) || timer_finished) {
-                progress();
+                progress(_test_opts.progress_count);
```

Comment:
Can `_test_opts.progress_count` replace `BUSY_PROGRESS_COUNT `?

Reply:
If we still don't want calculate the timer too soon, we should preserve BUSY_PROGRESS_COUNT. progress_count is 1 by default. it should be far less than BUSY_PROGRESS_COUNT.

Reply:
so maybe just set `progress_count = 1000` by default?

Reply:
I missed something earier. progress_count indicates the max number of ucp_worker_progress calls being made if there is still any CQEs left. In the case where no CQE left at all, it will behave the same way as it was before. So I guess it should have no impact.
1 by default is what Yossi suggests. The actual value should be dependent on real situation. from our testing experience, progress_count set to 1000 would harm performance badly.

Reply:
@yosefe but then we will have different behavior (and perf) for AM and TM mode, is this what we want?
I mean different iomsg handling which in case of AM is handled from CB, internal queues should be empty in normal flow